cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/system/SystemWorkerServiceLauncher.java

```diff
@@ -150,6 +153,24 @@ public void run() {
                     ArtifactLocalizerTwillRunnable.class.getSimpleName());
           }
 
+          if (twillPreparer instanceof SecureTwillPreparer) {
+            SecurityContext securityContext = createSecurityContext();
+            twillPreparer = ((SecureTwillPreparer) twillPreparer)
+                .withSecurityContext(SystemWorkerTwillRunnable.class.getSimpleName(), securityContext);
+            // Mount secret in ArtifactLocalizer sidecar which only run trusted code,
+            // so requests originated by ArtifactLocalizer can run with system identity when internal auth
+            // is enabled.
+            twillPreparer = ((SecureTwillPreparer) twillPreparer)
+                .withSecretDisk(ArtifactLocalizerTwillRunnable.class.getSimpleName(),
+                    new SecretDisk(cConf.get(Constants.Twill.Security.MASTER_SECRET_DISK_NAME),
+                        cConf.get(Constants.Twill.Security.MASTER_SECRET_DISK_PATH)));
+            String secretName = cConf.get(Constants.Twill.Security.WORKER_SECRET_DISK_NAME);
```

Comment:
Since both the `ArtifactLocalizer` and the `SystemWorker` containers are both running system code, they should probably both be using the `MASTER_SECRET_DISK_NAME` and `MASTER_SECRET_DISK_PATH`.

Reply:
Makes sense, addressed.